Python_classes_Bio_hw1_4.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+# Task 4 for fasta files manipulations. Подсчёт числа последовательностей в фаста файле
+#Построение гистограммы длин последовательностей
+#Подсчёт GC состава
+#Построение гистограммы частоты 4-меров (по оси x каждый из возможных 4-меров, а по y - их частота)
+#Переопределение метода для вывода информации при принте (достаточно текста с указанием путя к файлу)
+#Выполнение всех реализованных методов по подсчёту метрик
+#Можно придумать дополнительные метрики и реализовать их (по 1 баллу за каждую)
+
+
+# In[104]:
+
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+class FastaStat:
+    def __init__(self, path):
+        import Bio
+        from Bio import SeqIO
+        from Bio.SeqUtils import GC,molecular_weight
+        self.sequences = list(SeqIO.parse(open(path),'fasta'))
```

Comment:
`SeqIO.parse` принимает в том числе путь к файлу
```suggestion
        self.sequences = list(SeqIO.parse(path,'fasta'))
```
В исходном варианте он к тому же не закрывается